app/service/UserService.scala

```diff
@@ -41,5 +41,15 @@ class UserService @Inject() (
     } yield updated
 
   def getUser(id: String)(implicit hc: HeaderCarrier): Future[Option[Operator]] =
-    connector.getUserDetails(id)
+    for {
+      userDetails <- connector.getUserDetails(id).map(_.filterNot(_.deleted))
+    } yield userDetails
+
+  def markDeleted(user: Operator, operatorMakingTheChange: Operator)(
+    implicit hc: HeaderCarrier
+  ): Future[Operator] =
+    for {
+      deleted <- connector.markDeleted(user)
+      _ = auditService.auditUserUpdated(user, operatorMakingTheChange)
```

Comment:
I think this should provide `deleted` to the audit function

Reply:
Ah I see we only use the `id` of the operator in the audit payload so this probably doesn't matter

Reply:
yeah it uses the id, but I changed that anyway, good spot!